### PR TITLE
Multiline edit everywhere

### PR DESCRIPTION
**Pull request by Morb0**
## About the PR

Use new TextEdit for Papers, CommunicaitonConsole and AdminAnnounce.
Make some hacks because lack of events.
And rebind newline key function to Shift + Enter (most default thing?)
For coms console only 2 newlines allowed, other newlines replaced with space

**Media**

https://user-images.githubusercontent.com/14136326/230717501-e032214b-2e8f-43e8-9cde-2571be079065.mp4

image

image

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- tweak: Support multiline text edit for papers, communication console and admin announcements

### CODE REVIEW

File: Content.Client/Administration/UI/AdminAnnounceWindow.xaml.cs [hunk 1]
```diff
@@ -16,18 +18,20 @@ public AdminAnnounceWindow()
             RobustXamlLoader.Load(this);
             IoCManager.InjectDependencies(this);
 
+            var loc = IoCManager.Resolve<ILocalizationManager>();
+            Announcement.Placeholder = new Rope.Leaf(loc.GetString("admin-announce-announcement-placeholder"));
+
```
Comment (mirrorcult) [inline Content.Client/Administration/UI/AdminAnnounceWindow.xaml.cs:23]: this class has `ILocalizationManager` already injected (`_localization`)

**Review by mirrorcult — CHANGES_REQUESTED** [review]
paper should definitely use this but not sure about comms console / admin announcements -- people could spam newlines into them which might be annoying? (although realistically thats an admin issue and really easy to deal with)

**Comment by vulppine** [comment]
> paper should definitely use this but not sure about comms console / admin announcements -- people could spam newlines into them which might be annoying? (although realistically thats an admin issue and really easy to deal with)

could also filter newlines into spaces or limit the amount of newlines in an announcement

**Comment by mirrorcult** [comment]
yeah i'd probably limit the number of newlines

**Comment by Morb0** [comment]
> yeah i'd probably limit the number of newlines

Done, only 2 newlines [1984]

File: Content.Client/Communications/UI/CommunicationsConsoleBoundUserInterface.cs [hunk 2]
```diff
@@ -59,8 +55,22 @@ public void EmergencyShuttleButtonPressed()
 
         public void AnnounceButtonPressed(string message)
         {
-            var msg = message.Length <= 256 ? message.Trim() : $"{message.Trim().Substring(0, 256)}...";
-            SendMessage(new CommunicationsConsoleAnnounceMessage(msg));
+            var msg = (message.Length <= 256 ? message.Trim() : $"{message.Trim().Substring(0, 256)}...").ToCharArray();
+
+            // No more than 2 newlines, other replaced to spaces
```
Comment (DrSmugleaf) [inline Content.Client/Communications/UI/CommunicationsConsoleBoundUserInterface.cs:60]: This should be checked on the server as well, when it receives the CommunicationsConsoleAnnounceMessage (same for the 256 char limit)

**Review by DrSmugleaf — CHANGES_REQUESTED** [review]

**Review by DrSmugleaf — APPROVED** [review]

**Pull request merged**